tests/Unit/Core/ControllerFactoryTest.php

```
@@ -0,0 +1,116 @@
+    public function makeLeavesRuntimeConfigUntouchedWhenStoredProviderIsUnsupported(): void
+    {
+        $hasRowStmt = $this->createMock(\PDOStatement::class);
+        $hasRowStmt->method('fetch')->willReturn(['1' => 1]);
+        $settingsStmt = $this->createMock(\PDOStatement::class);
+        $settingsStmt->method('fetch')->willReturn([
+            'settings_json' => json_encode([
+                'ai_provider' => 'anthropic',
+                'ai_model' => 'claude-unknown',
+            ]),
+        ]);
+        $db = $this->getMockBuilder(Database::class)->disableOriginalConstructor()->getMock();
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $db. Configured minimum length is 3. (undefined)

(ShortVariable)